File: q3461/q3461.py
class Solution:
    def hasSameDigits(self, s: str) -> bool:
        n = len(s)
        digitList = [int(d) for d in s]
        for i in range(1, n - 1):
            for j in range(n - i):
                d1 = digitList[j]
                d2 = digitList[j+1]
                digitList[j] = (d1 + d2) % 10
        return digitList[0] == digitList[1]

File: q3461/q3461.java
class Solution {
    public boolean hasSameDigits(String s) {
        String currStr = doOperation(s);
        while (currStr.length() != 2) {
            currStr = doOperation(currStr);
        }
        return currStr.charAt(0) == currStr.charAt(1);
    }

    private String doOperation(String s) {
        if (s.length() == 2) return s;

        StringBuilder sb = new StringBuilder();
        for (int i = 1; i < s.length(); i++) {
            int res = ((int)s.charAt(i) + (int)s.charAt(i-1)) % 10;
            sb.append(res);
        }
        return sb.toString();
    }
}
